scalding-core/src/main/scala/com/twitter/scalding/Mode.scala

```diff
@@ -15,41 +15,53 @@ limitations under the License.
 */
 package com.twitter.scalding
 
-import java.io.File
-import java.util.{ UUID, Properties }
+import java.io._
+import java.lang.reflect.Constructor
+import java.nio.charset.Charset
+import java.nio.file.{ Files, Paths }
+import java.util.{ Properties, UUID }
 
 import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.{ FileSystem, Path }
-import org.apache.hadoop.mapred.JobConf
-
-import cascading.flow.{ FlowProcess, FlowConnector, FlowDef, Flow }
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapred.{ JobConf, OutputCollector, RecordReader }
+import cascading.flow._
 import cascading.flow.local.LocalFlowConnector
 import cascading.flow.local.LocalFlowProcess
+import cascading.flow.planner.BaseFlowStep
 import cascading.property.AppProps
-import cascading.tap.Tap
-import cascading.tuple.Tuple
-import cascading.tuple.TupleEntryIterator
+import cascading.scheme.NullScheme
+import cascading.tap.{ SinkMode, Tap }
+import cascading.tuple.{ Fields, Tuple, TupleEntryIterator }
+import com.twitter.maple.tap.MemorySourceTap
+import com.twitter.scalding.StorageMode.TemporarySource
+import com.twitter.scalding.filecache.{ CachedFile, UncachedFile }
+import com.twitter.scalding.reducer_estimation.ReducerEstimatorStepStrategy
+import com.twitter.scalding.typed.MemorySink
 
 import scala.annotation.tailrec
 import scala.collection.JavaConverters._
 import scala.collection.mutable.Buffer
 import scala.collection.mutable.{ Map => MMap }
 import scala.collection.mutable.{ Set => MSet }
 import scala.util.{ Failure, Success }
-
 import org.slf4j.LoggerFactory
 
+import scala.annotation.meta.param
+import scala.collection.{ Map, mutable }
```

Comment:
can we avoid `scala.collection.Map` and explicitly use `Map` for immutable map (standard) and `mutable.Map` when we need a mutable one?